pandas/core/arrays/arrow/array.py

```diff
@@ -2419,6 +2419,10 @@ def _str_rpartition(self, sep: str, expand: bool) -> Self:
     def _str_slice(
         self, start: int | None = None, stop: int | None = None, step: int | None = None
     ) -> Self:
+        if step is not None and step < 0:
+            # GH#59710
+            result = self._apply_elementwise(lambda x: x[start:stop:step])
+            return type(self)(pa.chunked_array(result))
         if start is None:
             start = 0
```

Comment:
See my comment in https://github.com/pandas-dev/pandas/issues/59710, but so probably this should be `start = -1` if `step < 0`, and then the above fallback might not be needed